app/scripts/contentscript.js

```diff
@@ -21,8 +23,19 @@ const inpageBundle = inpageContent + inpageSuffix
 // MetaMask will be much faster loading and performant on Firefox.
 
 if (shouldInjectProvider()) {
-  injectScript(inpageBundle)
-  start()
+  // If this is Brave, it requires coordination to know if we should be the
+  // web3 provider.
+  if (chrome.braveWallet && chrome.braveWallet.getWeb3Provider) {
+    chrome.braveWallet.getWeb3Provider((provider) => {
+      if (provider === extension.runtime.id) {
+        injectScript(inpageBundle)
```

Comment:
I agree with @Gudahtt that injecting async is asking for trouble, so I really recommend addressing that for Brave's sake. Since this doesn't introduce any async operation to our WebExtension-based injection, I'm okay with this.

Reply:
Yep I'll get it working sync.

Reply:
I'll add the typeof check on chrome too.